### ISSUE TITLE
3.0 - Reports use inconsistent type, status parameters to query orders

### ISSUE THREAD

**Issue by robincornett** [easydigitaldownloads/easy-digital-downloads#8735]
tl;dr: this started out as an enhancement request, but once I reset my store and started reviewing reports with very minimal data, I discovered that the report data is not consistent from one screen to the next. It's not clear if some numbers are supposed to be gross or net values, and some are definitely not correct at all, which is easiest to see with very little data.

Specifically, I recommend setting up a store like this, or at least this is my latest:
1. Have just one download and make the price an even number. Mine is $50.
2. Generate a few orders (I have three).
3. Do a partial refund on one order (I did a partial refund of $25).
4. Start looking over all the reports.

### Explain your enhancement (please be detailed)
It might be helpful to review the gateway sales reports table and headings, as I think they are not technically accurate.

### Justification or use case
When I was reviewing #8676, I was a little confused by the tables for one of my currencies (the currency isn't relevant but there were few enough orders that I noticed it):

![image](https://user-images.githubusercontent.com/4853066/124155906-ffe95a80-da64-11eb-92de-936cbab8577b.png)

So there are two orders which were placed with this currency. One has been fully refunded, and one has been partially refunded. I feel a little off balance looking at this table for a few reasons:
- Shouldn't the partially refunded order be included in the completed sales? Specifically, that order was for a download and had a fee on the order, and I refunded the fee, but not the download. So technically this still seems like a completed sale to me.
- The "Refunded Sales" heading feels wrong to me, because the query is not counting orders which were refunded; it is counting actual refunds which have been issued. So if I have one order, and issue two partial refunds on it, the "Total Sales" column will show `1`, but the "Refunded Sales" column will show `2`. (And "Complete Sales" will show `0`.) I feel like the heading should be "Refunds Issued" or something, as that would be more accurate with the data.

Additionally, I suspect the tiles are not showing accurate information. Here's what's actually in the database for these orders:
![image](https://user-images.githubusercontent.com/4853066/124157192-7f2b5e00-da66-11eb-898c-adf035cc0271.png)
- Sales: this matches the total number of sales.
- Earnings: I don't know if this is supposed to be a gross or net number, but it doesn't seem to be either. The gross would be `1062.75`, wouldn't it? And then net would be `29.99`?
- Refunds: This should be `1032.76`, I believe.
- Average Order Value: This seems correct based on gross numbers.

Updating to add:
It would be nice if the filters in the reports accepted some parameters. For example:

https://github.com/easydigitaldownloads/easy-digital-downloads/blob/977dd6ed110953168887bea92eb77847416b63db/includes/admin/reporting/reports.php#L1185-L1188

If anyone using this filter could also access the `$dates` and `$gateway` information that's part of the parameters sent to the `get_gateway_sales` method, it seems like that would be helpful. Seeing that consistently with the `apply_filters` calls here.

**Comment by robincornett** [comment]
Note: I think the tiles would be fixed fairly readily by explicitly setting `type` (or `type__in`) and rechecking the other parameters. The Earnings and Refunds tiles do not have a `type` set, so I think that accounts for some of the erroneous data. The Refunds tile should maybe be changed to `refund` for the `type`, or include `partially_refunded` in the `status` (right now it's just `refunded`).

I think the tiles are not right so I'm going to flag this as a bug.

**Comment by LisaCee** [comment]
I agree that the tiles are not right. From your example:
_Note: I'm using $ here instead of EU, because I'm lazy and it's easier_
- Sales: 2 sales - that makes sense and seems correct
- Earnings: I can't figure out the math here. If it's including tax I get $1062.75; the same as you. Without tax it would be $1059.98. I'm not sure which is right, but neither of them equal the $1032.76 shown on the tile.
- Refunds: I see where the $32.76 is coming from, but what about the fee? If the fee's not included here then it probably shouldn't be included in Earnings. Or, my preference, the fee should be included in Refunds.
- Average: I'm not clear on the math here, either. I would think it's (Earnings - Refunds) / Sales ($500) or Earnings / Sales ($516.38)

**Comment by ashleyfae** [comment]
> It would be nice if the filters in the reports accepted some parameters. For example:

Let's not do this in this issue. I want to remove those filters if they don't exist in 2.x. See https://github.com/easydigitaldownloads/easy-digital-downloads/issues/8463

**Comment by ashleyfae** [comment]
> I think we need to review a lot of the reports, actually--I reset my store to simplify what I'm looking at and this, for example, is what I have for the Downloads report:

That data actually doesn't come from reports. If you look at `class-top-selling-downloads-list-table.php`, it's actually coming from the download sales/earnings meta, which gets incremented on each sale. So I think that should be examined in a separate issue, since it's not actually reports data.
